bin/main.rb

```diff
@@ -1,2 +1,76 @@
 #!/usr/bin/env ruby
-puts 'Hello world!'
+
+puts 'WELCOME TO TIC-TAC-TOE GAME !!!'
+
+# create an empty game board
+@board = %w[1 2 3 4 5 6 7 8 9]
+
+# create game tokens as an array of 2 items
+@game_tokens = %w[X O]
+
+# create an empty hash to hold player and token info for each player
+# that is, player_one: 'X', player_two: 'O' or vice versa
+@assigned_tokens = {}
+
+# register two players
+puts 'Player 1, input name: '
+@player_one = gets.chomp
+
+puts 'Player 2, input name: '
+@player_two = gets.chomp
+
+# allow first player choose a game token X or O
+puts "#{@player_one}, Choose a token: 'X' or 'O'"
+selected_token = gets.chomp
+while selected_token
+  selected_token = selected_token.upcase
+
+  # validate the token chosen
+  if !@game_tokens.to_s.include? selected_token
+    puts "Invalid token, please select 'X' or 'O'"
+  else
+    # assign the selected_token to player_one and the other token to player_two
+    @assigned_tokens[:player_one] = selected_token
+    @assigned_tokens[:player_two] = if selected_token == @game_tokens[0]
+                                      @game_tokens[1]
+                                    else
+                                      @game_tokens[0]
+                                    end
+    puts 'Tokens assigned successfully'
+    puts "#{@player_one}'s Token:  #{@assigned_tokens[:player_one]}"
+    puts "#{@player_two}'s Token:  #{@assigned_tokens[:player_two]}"
+    break
+  end
+end
+
+game_on = true
+while game_on
+  # display gameboard with available moves
+  # prompt current player to select a move which is an index
+  puts 'Enter a position: '
+  position = gets.chomp
+  puts position
+
+  # validate the position. If already taken or if out of available board index,
+  # display message and prompt for the input again.
+
+  # if position is valid, update board
+  board[index] = currentPlayersToken
```

Comment:
- [ ] [REQUIRED] The game ends abruptly when a position is chosen. It should display the board after the selection and repeat all actions for the next player. Let's work on this.
<img width="726" alt="Screen Shot 2020-10-01 at 4 51 30 PM" src="https://user-images.githubusercontent.com/18187039/94840295-ba313000-0407-11eb-9419-89cbcb022cb9.png">